### PR TITLE
DEV: Display fuzzy site setting search results below direct matches

### PR DESCRIPTION
**Pull request by oblakeerickson**
When searching for site settings, in the results under the ALL category
all the fuzzy search results were showing first followed by any direct
matches. This change adjusts that so that fuzzy searches show below
direct matches.

Fuzzy results are now also sorted based on their gap calculation in
ascending order.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/controllers/admin-site-settings.js [hunk 1]
```diff
@@ -22,6 +22,17 @@ export default class AdminSiteSettingsController extends Controller {
     return 100;
   }
 
+  sortSettings(settings) {
+    // Sort the site settings so that fuzzy results are at the bottom
+    // and ordered by their gap count asc.
+    settings.sort((a, b) => {
+      const aSort = a.sort === undefined ? 0 : a.sort;
```
Comment (martin-brennan) [inline app/assets/javascripts/admin/addon/controllers/admin-site-settings.js:29]: I think just having this called `.sort` is a little confusing; something like `.weight` or `.sortWeight` would be better IMO

File: app/assets/javascripts/admin/addon/controllers/admin-site-settings.js [hunk 2]
```diff
@@ -22,6 +22,17 @@ export default class AdminSiteSettingsController extends Controller {
     return 100;
   }
 
+  sortSettings(settings) {
+    // Sort the site settings so that fuzzy results are at the bottom
+    // and ordered by their gap count asc.
+    settings.sort((a, b) => {
```
Comment (martin-brennan) [inline app/assets/javascripts/admin/addon/controllers/admin-site-settings.js:28]: You can just `return settings.sort` here

File: app/assets/javascripts/admin/addon/controllers/admin-site-settings.js [hunk 3]
```diff
@@ -86,6 +99,11 @@ export default class AdminSiteSettingsController extends Controller {
               strippedSetting.length <=
               strippedQuery.length + fuzzySearchLimiter
             ) {
+              const gapResult = strippedSetting.match(fuzzyRegexGaps);
+              if (gapResult) {
+                const filteredGapResult = gapResult.filter((gap) => gap !== "");
+                item.sort = filteredGapResult.length;
```
Comment (martin-brennan) [inline app/assets/javascripts/admin/addon/controllers/admin-site-settings.js:105]: Could probably be a one-liner, also same comment about `.sort` -> `.weight`

**Review by martin-brennan — COMMENTED** [review]
Nice one, the `digest` example especially is working WAY better now!

File: app/assets/javascripts/admin/addon/controllers/admin-site-settings.js (hunk 1 above)
Comment (oblakeerickson) [reply app/assets/javascripts/admin/addon/controllers/admin-site-settings.js:29]: Ah yes, thanks for the name suggestion, always appreciated.

**Review by martin-brennan — APPROVED** [review]
Thanks for the change Blake!

**Pull request merged**